### DISCUSSION TITLE
Getting "unknown file extension .ts" error on fresh vite-plugin-ssr React + TS  project

### DISCUSSION THREAD

**Discussion by alok8bb** [vikejs/vike#1067] [Discussion] [answered]
Hello, I wanted to try vite-plugin-ssr so I created the new project using  `npm init vite-plugin-ssr@latest` but after installing the dependencies and running `pnpm run dev` I'm getting this error. 
![image](https://github.com/brillout/vite-plugin-ssr/assets/73450969/e9be09d1-92df-43a0-900d-58e074580d8c)

**Comment by alok8bb** [comment]
https://github.com/esbuild-kit/tsx using this worked, its ts-node issue

**Comment by brillout** [comment]
I couldn't reproduce.

Have you seen https://github.com/pnpm/pnpm/issues/6463? It may be related to that.

Reproduction that includes a `pnpm-lock.yaml` welcome.

**Reply by alok8bb** [reply]
I tried with npm, got the same error: 
https://bin.mkr.pw/~64d883263f040d14f3e985ee

pacakge-lock.json - https://gist.github.com/alok8bb/4843104ec17a30b7952933d28cb8b6d3
package.json - https://gist.github.com/alok8bb/218f507bbcc32e27d1252a34c689c540

**Reply by brillout** [reply]
What's your Node.js version?

**Reply by brillout** [reply]
Nevermind, I just saw your screenshot.

**Reply by brillout** [reply]
That's quite unexpected, I'll have a look at it.

**Reply by brillout** [reply]
I could reproduce with Node.js `20.4.0`.

**Reply by brillout** [reply, marked as answer]
Actually, it's a duplicate of https://github.com/brillout/vite-plugin-ssr/issues/877. I just updated the boilerplates to use [this workaround](https://github.com/TypeStrong/ts-node/issues/1997#issuecomment-1537111441): b77acfda33752585469ee79b402e22bddef7b23c.

Thanks for the reporting this.

Btw. in case your company is up for it: https://github.com/sponsors/brillout.
